### PR TITLE
Addition of CMake presets for common build configurations + Conan automation

### PR DESCRIPTION
**Pull request by piponazo**
In this branch I am exploring a recent CMake feature named "Presets" which was added on CMake 3.20:

https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html

These presets enables the usage of configurations composed by many CMake variables in an easy way. Instead of passing several of these options with the CMake argument `-D` one can specify the preset to use with the argument `--preset`. Another advantage of using such presets, is that C++ IDEs such as "Visual Studio", "Visual Studio Code", "CLion", "XCode", etc. can benefit from them and allow a better integration with CMake projects.

I am also exploring how to automate the usage of conan thanks to the presets. In that way, we would not need to run the following steps manually:
1. Create build directory
2. Run conan to bring dependencies
3. Run CMake to configure the project with certain options

Instead, by using a preset, these 3 steps would be done at once.

### CODE REVIEW

**Comment by kmilos** [comment]
Nice! MSYS2 has CMake 3.23 for a while, so it should be working there. Might want to install ninja though, maybe that was the problem...

**Comment by kmilos** [comment]
I.e., the folowing should in theory allow you to switch MSYS2 also:

```
diff --git a/.github/workflows/on_PR_windows_matrix.yml b/.github/workflows/on_PR_windows_matrix.yml
index 13cf3acc..a7b53694 100644
--- a/.github/workflows/on_PR_windows_matrix.yml
+++ b/.github/workflows/on_PR_windows_matrix.yml
@@ -122,17 +122,21 @@ jobs:
           install: >-
             base-devel
           pacboy: >-
-            toolchain:p
+            cc:p
+            gcc-libs:p
+            libwinpthread:p
             cmake:p
+            ninja:p
+            gtest:p
+            curl:p
             expat:p
             gettext:p
-            gtest:p
             libiconv:p
             zlib:p
 
       - name: Build
         run: |
-          cmake -G"MSYS Makefiles" \
+          cmake -GNinja \
             -DCMAKE_CXX_FLAGS=-Wno-deprecated \
             -DCMAKE_BUILD_TYPE=${{matrix.build_type}} \
             -DBUILD_SHARED_LIBS=${{matrix.shared_libraries}} \
```

(The MSYS2 project is now recommending to install only what is minimally needed to build in order to reduce CI traffic and time; `toolchain` is convenient, but bloated...)

**Comment by piponazo** [comment]
@kmilos Thanks for the feedback! Indeed, I saw that cygwin had an older version of CMake and I quickly assumed that in MSYS the problem would be the same. I followed your suggestions and I will try to use again the CMake presets on MSYS and I also removed the usage of `toolchain` in favor of the individual packages.

It is also very nice to see that with the CMake presets, one can simply open the project in "Visual Studio" or "Visual Studio Code" by just opening the Exvi2 source code folder and choosing the appropriate preset.

**Comment by kmilos** [comment]
Ah, now it's failing because we now have switched on EXIV2_TEAM_WARNINGS_AS_ERRORS for MSYS2 from the common preset... Sorry, but might take me while until I manage to dig into that, but you can perhaps turn it off temporarily on Windows?

Re Cygwin - it should have CMake 3.20 as well, but maybe it was failing before because `libcurl-devel` was not installed, but you have it enabled in the common preset now. But it may likely fail again because of that warning switch again...

**Comment by piponazo** [comment]
> Ah, now it's failing because we now have switched on EXIV2_TEAM_WARNINGS_AS_ERRORS for MSYS2 from the common preset... Sorry, but might take me while until I manage to dig into that, but you can perhaps turn it off temporarily on Windows?

Right, I need to disable the warnings-as-errors there, it was disabled before.

> Re Cygwin - it should have CMake 3.20

The problem is that we are using the schema version 3 which has features which were introduced in CMake 3.21:
image

I will leave the Cygwin CI job as it is for the moment.

**Comment by kmilos** [comment]
> I will leave the Cygwin CI job as it is for the moment.

Sure, I just saw 3.20 in the conversation above and assumed that was the minimum...

Still, it doesn't hurt to add `libcurl-devel` and `-DEXIV2_ENABLE_CURL` to Cygwin in anticipation ;)

**Comment by piponazo** [comment]
I will still leave this PR as a draft because I want to explore the usage of the Presets on Linux and Mac, but at the moment I do not have access to such environments. I will let you know when I am happy about the status of this branch.

File: .gitignore [hunk 1]
```diff
@@ -25,5 +25,6 @@ doc/html
 contrib/vms/.vagrant
 /.vscode
 .vs/
+ CMakeUserPresets.json
```
Comment (kmilos) [inline .gitignore:28]: remove leading space?

File: CMakePresets.json [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      "displayName": "Base preset for macOS with default compiler: AppleClang (specially useful for CI jobs)",
+      "inherits": "base_ninja",
+      "condition": {
+        "type": "equals",
+        "lhs": "${hostSystemName}",
+        "rhs": "Darwin"
+      },
+      "cacheVariables": { 
+        "CMAKE_BUILD_TYPE": "Release",
+				"CONAN_AUTO_INSTALL": false
```
Comment (kmilos) [inline CMakePresets.json:61]: fix indentation

**Review by kmilos — DISMISSED** [review]
LGTM, just whitespace comments

**Review by kmilos — APPROVED** [review]

**Pull request merged**